src/plugins/list/list.c

```diff
@@ -0,0 +1,365 @@
+		}	
+		keySetBaseName(lookup, "get");
+		sub = ksLookup(cutKS, lookup, 0);
+		if(sub)
+		{
+			const char *getString = keyString(sub);
+			const char *getStrings[] = {"pregetstorage", "postgetstorage"};
+			GetPlacements getPlacement = preGetStorage;
+			while(getPlacement != getEnd)
+			{	
+				if(strstr(getString, getStrings[getPlacement]))
+				{
+					ksAppendKey(placements->getKS[getPlacement], keyDup(cur));
+				}
+				++getPlacement;
+			}
+		}	
+		keySetBaseName(lookup, "err");
+		sub = ksLookup(cutKS, lookup, 0);	
+		if(sub)
+		{
+
+			const char *errString = keyString(sub);
+			const char *errStrings[] = {"preRollback", "postRollback"};
+			ErrPlacements errPlacement = preRollback;
+			while(errPlacement != errEnd)
+			{	
+				if(strstr(errString, errStrings[errPlacement]))
+				{
+					ksAppendKey(placements->errKS[errPlacement], keyDup(cur));
+				}
+				++errPlacement;
+			}
+		}
+		keyDel(lookup);
+	}
+	ksDel(cutKS);
+	ksDel(config);
+	placements->initialized = 1;
+	return 1; /* success */
+}
+
+int elektraListClose(Plugin *handle , Key *errorKey ELEKTRA_UNUSED)
+{
+	/* free all plugin resources and shut it down */
+	Placements *placements = elektraPluginGetData(handle);
+	ksDel(placements->getKS[0]);
+	ksDel(placements->getKS[1]);
+	ksDel(placements->setKS[0]);
+	ksDel(placements->setKS[1]);
+	ksDel(placements->setKS[2]);
+	ksDel(placements->errKS[0]);
+	ksDel(placements->errKS[1]);
+	elektraFree(placements);
+	return 1; /* success */
+}
+
+Key *(*traversalFuntion)(const KeySet *);
+
+static int runPlugins(KeySet *pluginKS, KeySet *configOrig, KeySet *returned, Key *parentKey, OP op)
+{
+	Key *current;
+	Key *errorKey = parentKey;
+	KeySet *modules = ksNew(0, KS_END);
+	elektraModulesInit(modules, NULL);
+	while((current = ksNext(pluginKS)) != NULL)
+	{
+		const char *name = keyString(current);
+		elektraPluginFactory pluginFactory = 0;
+		pluginFactory = elektraModulesLoad(modules, name, errorKey);
+	        if(pluginFactory == NULL)
+		{
+			goto Error;
+		}		
+		Plugin *slave;
+		slave = pluginFactory();
+		if(slave == NULL)
+			goto Error;
+		Key *userCutPoint = keyNew("user", 0);
+		Key *sysConfCutPoint = keyNew("system", 0);
+		KeySet *config = ksDup(configOrig);
+		KeySet *sysConfigAll = ksCut(config, sysConfCutPoint);
+		KeySet *userConfigAll = ksCut(config, userCutPoint);
+		KeySet *pluginConfig = ksCut(userConfigAll, current);
+		KeySet *realPluginConfig = elektraRenameKeys(pluginConfig, "user");
+		ksDel(pluginConfig);
+		Key *toRemove = keyNew("user/plugins", 0);
+		ksDel(ksCut(sysConfigAll, toRemove));
+		ksAppend(realPluginConfig, sysConfigAll);
+		keyDel(toRemove);
+		toRemove = keyNew("user/placements",0);
+		ksDel(ksCut(realPluginConfig, toRemove));
+		ksRewind(realPluginConfig);
+		ksDel(sysConfigAll);
+		ksDel(userConfigAll);
+		ksDel(config);
+		keyDel(userCutPoint);
+		keyDel(sysConfCutPoint);
+		keyDel(toRemove);
+		slave->refcounter = 1;
+		slave->config = ksDup(realPluginConfig);
+		if(slave->kdbOpen)
```

Comment:
The plugin will be loaded every time it's needed and closed immediately after again.